internal/k8s/controllers/config_controller.go

```diff
@@ -92,18 +94,28 @@ func (r *ConfigReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 		return ctrl.Result{}, err
 	}
 
-	metallbCRs := config.ClusterResources{
+	var passwordSecrets corev1.SecretList
+	if err := r.List(ctx, &passwordSecrets, client.InNamespace(r.Namespace)); err != nil {
+		level.Error(r.Logger).Log("controller", "ConfigReconciler", "error", "failed to get secrets", "error", err)
+		return ctrl.Result{}, err
+	}
+
+	metallbCRsAndSecrets := config.ClusterResources{
 		Pools:              ipPools.Items,
 		Peers:              bgpPeers.Items,
 		BFDProfiles:        bfdProfiles.Items,
 		L2Advs:             l2Advertisements.Items,
 		BGPAdvs:            bgpAdvertisements.Items,
 		LegacyAddressPools: addressPools.Items,
 	}
+	metallbCRsAndSecrets.PasswordSecrets = make(map[string]corev1.Secret)
```

Comment:
I'd wrap this (and the List above) in a func to put in the bottom of the file, and then use the result when you do the literal initializer above.